dags/atd_finance_data_subprojects.py

```diff
@@ -0,0 +1,154 @@
+    "on_failure_callback": task_fail_slack_alert,
+}
+
+# Data tracker Knack app secrets
+DATA_TRACKER_SECRETS = {
+    "KNACK_APP_ID": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": "production.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "Knack AMD Data Tracker",
+        "opfield": "production.apiKey",
+    },
+}
+
+# Finance purchasing Knack app secrets
+FINANCE_PURCHASING_SECRETS = {
+    "KNACK_APP_ID": {
+        "opitem": "Knack Finance and Purchasing",
+        "opfield": "production.appId",
+    },
+    "KNACK_API_KEY": {
+        "opitem": "Knack Finance and Purchasing",
+        "opfield": "production.apiKey",
+    },
+}
+
+OTHER_SECRETS = {
+    "USER": {
+        "opitem": "Finance Data Warehouse Oracle DB",
+        "opfield": "production.Username",
+    },
+    "PASSWORD": {
+        "opitem": "Finance Data Warehouse Oracle DB",
+        "opfield": "production.Password",
+    },
+    "HOST": {
+        "opitem": "Finance Data Warehouse Oracle DB",
+        "opfield": "production.Host",
+    },
+    "PORT": {
+        "opitem": "Finance Data Warehouse Oracle DB",
+        "opfield": "production.Port",
+    },
+    "SERVICE": {
+        "opitem": "Finance Data Warehouse Oracle DB",
+        "opfield": "production.Service",
+    },
+    "BUCKET": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.Bucket",
+    },
+    "AWS_ACCESS_KEY_ID": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.Access ID",
+    },
+    "AWS_SECRET_ACCESS_KEY": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.Secret Access Key",
+    },
+    "TASK_DATASET": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.Task Dataset ID",
+    },
+    "FDU_DATASET": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.FDU Dataset ID",
+    },
+    "DEPT_UNITS_DATASET": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.Department Units Dataset ID",
+    },
+    "SUBPROJECTS_DATASET": {
+        "opitem": "atd-finance-data",
+        "opfield": "production.Subprojects Dataset ID",
+    },
+    "SO_USER": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeyId",
+    },
+    "SO_PASS": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.apiKeySecret",
+    },
+    "SO_TOKEN": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.appToken",
+    },
+    "SO_WEB": {
+        "opitem": "Socrata Key ID, Secret, and Token",
+        "opfield": "socrata.endpoint",
+    },
+}
+
+# Combine env vars to create one for each Knack app
+DATA_TRACKER_SECRETS.update(OTHER_SECRETS)
+FINANCE_PURCHASING_SECRETS.update(OTHER_SECRETS)
+
+with DAG(
+    dag_id="atd_finance_data_subprojects",
+    description="Gets Finance data from a database, places it in an S3 bucket, then moves it along to Knack and socrata.",
```

Comment:
I am tempted to say the descriptions should not be the same for all the dags, but when I went to the airflow UI to see where we see the descriptions, they are always with the dag_id, so maybe that gives enough context.

Reply:
i had the same thought, but I agree that the DAG names describe what data is being handled